### PR TITLE
Extract to chain config

### PR DESCRIPTION
**Pull request by mercepluka**

### CODE REVIEW

**Review by ijerkovic — APPROVED** [review]

File: chain/epoch-manager/src/reward_calculator.rs [hunk 1]
```diff
@@ -16,24 +16,25 @@ pub struct RewardCalculator {
...
     pub online_max_threshold: Rational32,
     pub num_seconds_per_year: u64,
+    pub chain_config_store: ChainConfigStore,
 }
 
 impl RewardCalculator {
     pub fn new(config: &GenesisConfig) -> Self {
+        let chain_config_store = config.clone().chain_config_store;
         RewardCalculator {
             max_inflation_rate: config.max_inflation_rate,
```
Comment (mhalambek) [inline chain/epoch-manager/src/reward_calculator.rs:30]: It looks to me that some of other parameters should be read from chain_config_store as well
max_inflation_rate like on of those

File: chain/epoch-manager/src/reward_calculator.rs [hunk 2]
```diff
@@ -57,7 +59,7 @@ impl RewardCalculator {
         let protocol_reward_rate = if use_hardcoded_value {
             Rational32::new_raw(1, 10)
         } else {
-            self.protocol_reward_rate
+            self.chain_config_store.get_config(epoch_height).protocol_reward_rate
```
Comment (mhalambek) [inline chain/epoch-manager/src/reward_calculator.rs:62]: Is it possible to store a reference to epoch height inside the chain_config_store itself,
This way chain_config_store would be able to determine epoch_height on it's own

**Review by mhalambek — APPROVED** [review]
Have some minor comments about the implementation, PR looks very solid otherwise

File: core/chain-configs/src/config_store.rs [hunk 3]
```diff
@@ -29,6 +29,9 @@ pub struct ChainConfigStore {
 }
 
 impl ChainConfigStore {
+    /// Constructs a new store. Initial chain config is read from provided genesis_config_snapshot.
+    /// Store values are read from config path list file and specific patches are read from the
```
Comment (ijerkovic) [inline core/chain-configs/src/config_store.rs:33]:
```suggestion
    /// Stored values are read from config path list file and specific patches are read from the
```

**Pull request closed without merging**